--- a/tao/src/platform_impl/ohos/mod.rs
+++ b/tao/src/platform_impl/ohos/mod.rs
@@ -51,7 +51,6 @@
 pub struct EventLoop<T: 'static> {
   pub(crate) openharmony_app: OpenHarmonyApp,
   window_target: event_loop::EventLoopWindowTarget<T>,
-  cause: StartCause,
   user_events_sender: mpsc::Sender<T>,
   user_events_receiver: PeekableReceiver<T>,
   event_handler: RefCell<Option<Box<dyn FnMut(event::Event<T>)>>>,
@@ -84,13 +83,11 @@
       window_target: event_loop::EventLoopWindowTarget {
         p: EventLoopWindowTarget {
           app: openharmony_app.clone(),
-          control_flow: Cell::new(ControlFlow::default()),
           exit: Cell::new(false),
           _marker: PhantomData,
         },
         _marker: PhantomData,
       },
-      cause: StartCause::Init,
       user_events_sender,
       user_events_receiver: PeekableReceiver::from_recv(user_events_receiver),
       event_handler: RefCell::new(None),
@@ -267,9 +264,6 @@
           }
         },
       },
-      _ => {
-        warn!("Unknown openharmony_ability input event {event:?}")
-      }
     }
   }

@@ -475,7 +469,6 @@
 #[derive(Clone)]
 pub struct EventLoopWindowTarget<T: 'static> {
   pub(crate) app: OpenHarmonyApp,
-  control_flow: Cell<ControlFlow>,
   exit: Cell<bool>,
   _marker: std::marker::PhantomData<T>,
 }
--- a/tao/src/window.rs
+++ b/tao/src/window.rs
@@ -1671,7 +1671,7 @@
   West,
 }

-#[cfg(any(target_os = "windows", target_os = "linux"))]
+#[cfg(any(target_os = "windows", all(target_os = "linux", not(target_env = "ohos"))))]
 pub(crate) fn hit_test(
   (left, top, right, bottom): (i32, i32, i32, i32),
   cx: i32,
--- a/tauri/crates/tauri-runtime-wry/src/lib.rs
+++ b/tauri/crates/tauri-runtime-wry/src/lib.rs
@@ -2988,7 +2988,7 @@
   }

   #[cfg(target_env = "ohos")]
-  fn new_any_thread(args: RuntimeInitArgs) -> Result<Self> {
+  fn new_any_thread(_args: RuntimeInitArgs) -> Result<Self> {
     unimplemented!()
   }

@@ -4006,6 +4006,8 @@
             }
           },
           WebviewMessage::WithWebview(f) => {
+            #[cfg(target_env = "ohos")]
+            let _ = f;
             #[cfg(all(
               any(
                 target_os = "linux",
--- a/tauri/crates/tauri/src/manager/webview.rs
+++ b/tauri/crates/tauri/src/manager/webview.rs
@@ -6,9 +6,11 @@
   borrow::Cow,
   collections::{HashMap, HashSet},
   fmt,
-  fs::create_dir_all,
   sync::{Arc, Mutex, MutexGuard},
 };
+
+#[cfg(not(any(target_os = "android", target_env = "ohos")))]
+use std::fs::create_dir_all;

 use serde::Serialize;
 use serialize_to_javascript::{default_template, DefaultTemplate, Template};
--- a/tauri/crates/tauri/src/app.rs
+++ b/tauri/crates/tauri/src/app.rs
@@ -1162,6 +1162,7 @@
         rx.recv().unwrap_or(false)
       }

+      /// Returns whether this platform supports multiple windows.
       #[cfg(target_env = "ohos")]
       pub fn supports_multiple_windows(&self) -> bool {
         false
